content/logs/languages/php.md

```diff
@@ -0,0 +1,562 @@
+
+        return $record;
+      }
+
+      protected function clean($array)
+      {
+        $toReturn = array();
+        foreach(array_keys($array) as $key) {
+          if (false !== strpos($key, 'password')) {
+            //Do not add
+          } else if (false !== strpos($key, 'csrf_token')) {
+            //Do not add
+          } else {
+            $toReturn[$key] = $array[$key];
+          }
+        }
+
+        return $toReturn;
+      }
+    }
+    ```
+
+2. Wire the processor with Symfony:  
+  ```yaml
+   services:
+      monolog.processor.session_request:
+          class: Acme\Bundle\MonologBundle\Log\SessionRequestProcessor
+          arguments:  [ @session ]
+          tags:
+              - { name: monolog.processor, method: processRecord }
+  ```
+
+3. [Stream generated JSON file to Datadog](/logs/#custom-log-collection)
+
+## PHP Zend-Log
+
+Zend-log is a part of the Zend framework. Use [Composer](https://getcomposer.org/) to add Zend-Log:
+
+```
+composer require "zendframework/zend-log"
+```
+
+Alternatively, install it manually:
+
+1. Download the source from the repository and include it to the libraries
+2. Initialize the instance in the application's bootstrap
+
+```php
+<?php
+require __DIR__ . '/vendor/autoload.php';
+
+use Zend\Log\Logger;
+use Zend\Log\Writer\Stream;
+use Zend\Log\Formatter\JsonFormatter;
+```
+
+### Setup - Log into files with Zend-log
+
+The following configuration enables the JSON formatting and writes the logs and events into the `application-json.log` file. Edit your code, right after the initialization of the Zend-Log instance and add a new handler.
+
+```php
+<?php
+
+use Zend\Log\Logger;
+use Zend\Log\Writer\Stream;
+use Zend\Log\Formatter\JsonFormatter;
+
+  
+// create a logger
+$logger = new Logger();
+
+// create a writer
+$writer = new Stream('file://' . __DIR__ . '/application-json.log');
+
+// create a Json formatter
+$formatter = new JsonFormatter();
+$writer->setFormatter($formatter);
+
+// bind
+$logger->addWriter($writer);
+Zend\Log\Logger::registerErrorHandler($logger);
+```
+
+Then [Stream your log files to Datadog](/logs/#custom-log-collection)
+
+### Add meta field and context
+
+Much of the usefulness information comes from additional context data that you can add to your logs and your events. Zend-Log makes this very convenient by providing methods to set thread local context data that is then submitted automatically with all events. At any moment, log an event with a contextual data:  
+
+```php
+$logger->info('Adding a new user', array('username' => 'Seldaek'));
+```
+
+But, most important, the library comes with a processor feature. Processors allow you to provide additional information to logs in an automated fashion. They are called from the logger before the event is passed to the writers; they receive the event array, and return an event array on completion.
+
+Use cases include:
+
+* Providing exception backtrace information.
```

Comment:
These first two bullet points have a period, but the last one does not. I prefer having no periods after bullet points, so I'd remove the periods from the first two.